src/librustc_typeck/check/mod.rs

```diff
@@ -1066,6 +1066,13 @@ fn check_fn<'a, 'gcx, 'tcx>(inherited: &'a Inherited<'a, 'gcx, 'tcx>,
         fcx.check_pat_walk(&arg.pat, arg_ty,
             ty::BindingMode::BindByValue(hir::Mutability::MutImmutable), true);
 
+        // If any of a function's parameters have a type that is uninhabited, then it
+        // may never be called (because its arguments cannot be constructed). Therefore,
```

Comment:
Nit: I would say "then it can never be called". "may never" sounds to me like it *might not* be called, but it still *could be*